iceberg/iceberg-handler/src/test/queries/positive/query_iceberg_virtualcol.q

```diff
@@ -2,8 +2,17 @@ drop table if exists tbl_ice;
 create external table tbl_ice(a int, b string) partitioned by (c int) stored by iceberg stored as orc tblproperties ('format-version'='2');
 insert into tbl_ice values (1, 'one', 50), (2, 'two', 50), (3, 'three', 50), (4, 'four', 52), (5, 'five', 54), (111, 'one', 52), (333, 'two', 56);
 
+select tbl_ice.ROW__POSITION from tbl_ice order by tbl_ice.ROW__POSITION;
+
 select a, c, tbl_ice.PARTITION__SPEC__ID, tbl_ice.PARTITION__HASH, tbl_ice.ROW__POSITION from tbl_ice
 order by tbl_ice.PARTITION__HASH, tbl_ice.ROW__POSITION desc;
 
 select a, c, tbl_ice.PARTITION__SPEC__ID, tbl_ice.PARTITION__HASH, tbl_ice.ROW__POSITION from tbl_ice
 sort by tbl_ice.PARTITION__HASH, tbl_ice.ROW__POSITION desc;
+
+
+-- create a table with more than 4 columns
+create table ice01 (a int, b int, c int, d int, e int) stored by iceberg tblproperties ('format-version'='2');
+insert into ice01 values (1,2,3,4,5), (6,7,8,9,10);
+
+select ice01.ROW__POSITION from ice01 order by ice01.ROW__POSITION;
```

Comment:
Without your PR, I also successfully ran the query in my local Hive which is latest master code. I can not reproduce the failed case described in HIVE-27562.
Could you please check if it is not a bug?

Reply:
I tried again and it reproduced, may be the order by clause which I added here for the sake of test is doing something. I ran the current test reverting the prod fix & it failed.
```
[ERROR] Failures: 
[ERROR]   TestIcebergCliDriver.testCliDriver:61 Client execution failed with error code = 1 
running 

select tbl_ice.ROW__POSITION from tbl_ice 
fname=query_iceberg_virtualcol.q

See ./ql/target/tmp/log/hive.log or ./itests/qtest/target/tmp/log/hive.log, or check ./ql/target/surefire-reports or ./itests/qtest/target/surefire-reports/ for specific test cases logs.
 java.lang.RuntimeException: java.io.IOException: java.lang.IndexOutOfBoundsException: start index (4) must not be greater than size (3)
        at org.apache.hadoop.hive.ql.exec.FetchTask.executeInner(FetchTask.java:210)
        at org.apache.hadoop.hive.ql.exec.FetchTask.execute(FetchTask.java:95)
        at org.apache.hadoop.hive.ql.Driver.runInternal(Driver.java:212)
        at org.apache.hadoop.hive.ql.Driver.run(Driver.java:154)
        at org.apache.hadoop.hive.ql.Driver.run(Driver.java:149)

```

Reply:
ACK. I removed the `order by` and then reproduced the failed case. Thanks for explanation.

Reply:
`set hive.fetch.task.conversion=none;`
Just curious, I tested with launching Tez job instead of fetch task, and i found  the query can ran successfully without the PR.
I have not explored the difference, but it seems interesting.

Reply:
hmm, I didn't try, it is working that is good only :-)